kernel/mmanutils.c

```diff
@@ -261,6 +276,63 @@ void* myst_mmap(
         return addr;
     }
 
+    /* Handle shared mappings */
+    shared_mapping_t* new_sm = NULL;
+    if (fd >= 0 && (flags & MAP_SHARED))
+    {
+        // check if there is existing shared mapping
+        // if not add one
+        myst_file_t* file = NULL;
+        myst_fs_t* fs = NULL;
+        myst_fdtable_t* fdtable = myst_fdtable_current();
+        char* realpath = malloc(PATH_MAX);
+
+        if (!realpath)
+            return (void*)-1;
+
+        /* Get file path */
+        if (myst_fdtable_get_file(fdtable, fd, &fs, &file) < 0)
+        {
+            free(realpath);
+            return (void*)-1;
+        }
+        if ((*fs->fs_realpath)(fs, file, realpath, PATH_MAX) < 0)
+        {
+            free(realpath);
+            return (void*)-1;
+        }
+        myst_spin_lock(&_shared_mappings_lock);
+        shared_mapping_t* sm = (shared_mapping_t*)_shared_mappings.head;
+        while (sm)
+        {
+            if ((strcmp(realpath, sm->path) == 0) && offset == sm->offset &&
+                length == sm->length)
+            {
+                sm->nusers++;
+                myst_spin_unlock(&_shared_mappings_lock);
```

Comment:
Can we support write through here? According to the main page:
>        MAP_SHARED
              Share this mapping.  Updates to the mapping are visible to
              other processes mapping the same region, and (in the case
              of file-backed mappings) are carried through to the
              underlying file.